content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,33 @@
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you find things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you find things *from* the web (information and answers). Here are some examples `TODO: make it clear which these are available and which are future directions`:
+
+- Trying to figure out the best way to call a function? Code search could show you a semi-exhaustive list of call sites. Code intelligence is also being able to see how the code owners of that function themselves have recently been calling that function.
+- Need to debug an issue in code? Code search could help you navigate within the same project to understand it. Code intelligence is also being able to see what changed and why, and trace calls across projects and dependencies/dependents.
+- Want to fix or improve something across your entire codebase? Code search might help you find where needs fixing. Code intelligence is also being able to push up a quick fix everywhere, track its progress, and monitor for regressions.
+- Need to see how your team's API is being used? Code search could show you a list of file locations. Code intelligence is also seeing which other teams are using it and the trend of usage over time.
+
+A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Code hosts are too conflicted to make a good code intelligence platform because they want you to use *their* offerings in each category (and suffer vendor lock-in), but we're committed to Sourcegraph being universal so you can get full code intelligence *and* keep using the best tools in each category.
```

Comment:
I don't love explicitly saying "Code hosts are too conflicted to make a good code intelligence platform because they want you to use *their* offerings in each category (and suffer vendor lock-in), but we're committed to Sourcegraph being universal so you can get full code intelligence *and* keep using the best tools in each category." While this messaging is valid and can be used elsewhere, I'd recommend keeping the blog post about Sourcegraph. We can explain why we're in the best position to be a code intelligence platform vs saying why others can't.

Reply:
+1, and I think cutting that and jumping straight to the context about being universal will still get that point across

Reply:
```suggestion
A code intelligence platform includes code search *and* information about your code from so many other tools, etc., that you rely on: code ownership (and mapping of that to teams, on-call, etc.), code history and reviews, the dependency graph, runtime service/API usage, code and security scanners, deployment status, CI pipeline output, etc. Sourcegraph is committed to being universal so you can get full code intelligence *and* keep using the best tools in each category.
```